File: src/main/java/org/hisp/dhis/rules/models/TimeInterval.java
package org.hisp.dhis.rules.models;

import org.joda.time.LocalDate;

public class TimeInterval
{
    private final LocalDate startDate;

    private final LocalDate endDate;

    private final boolean empty;

    private TimeInterval( LocalDate startDate, LocalDate endDate, boolean empty )
    {
        this.startDate = startDate;
        this.endDate = endDate;
        this.empty = empty;
    }

    public static TimeInterval empty()
    {
        return new TimeInterval( null, null, true );
    }

    public static TimeInterval fromTo( LocalDate startDate, LocalDate endDate )
    {
        return new TimeInterval( startDate, endDate, false );
    }

    public LocalDate getStartDate()
    {
        return startDate;
    }

    public LocalDate getEndDate()
    {
        return endDate;
    }

    public boolean isEmpty()
    {
        return this.empty;
    }
}

File: src/jvmMain/kotlin/org/dhis2/ruleengine/models/TimeInterval.kt
package org.dhis2.ruleengine.models

import org.joda.time.LocalDate

class TimeInterval private constructor(val startDate: LocalDate?, val endDate: LocalDate?, val isEmpty: Boolean) {

    companion object {
        fun empty(): TimeInterval {
            return TimeInterval(null, null, true)
        }

        fun fromTo(startDate: LocalDate?, endDate: LocalDate?): TimeInterval {
            return TimeInterval(startDate, endDate, false)
        }
    }
}